src/sync/server.rs

```diff
@@ -373,6 +374,9 @@ impl Server {
                         }
                         Err(e) => {
                             error!("listener accept got {:?}", e);
+                            // Since poll is level-triggered, an uncorrected error can lead to an infinite loop,
+                            // so we sleep for a while and wait for the error to be corrected.
+                            thread::sleep(Duration::from_secs(10));
```

Comment:
10 seconds seem like a long time?

Reply:
Yes, It's a long time for a program, but the errors of accept such as EMFILE and ENOMEM can't be recovered in a short time, then the logs will filled up the disk.

Reply:
I think I could add a method server.set_accept_retry_interval, so users can customize the sleep duration

Reply:
Hi, @jsturtevant I have added the method set_accept_retry_interval.

Reply:
Thanks making it configurable.

If it wasn't resolved in 10s wouldn't the continue loop still get stuck?  In either case, would EMFILE is eventually resolved, wouldn't the loop finish?

Is the sleep to resolve cpu cycles while those errors occur?  Should this only be done for a subclass of errors?  seems like if there was a transient error we would slow things down significantlly.

Reply:
I updated the code. Now the sleep will only affects those resource limit errors: libc::EMFILE, libc::ENFILE, libc::ENOBUFS, libc::ENOMEM.

Reply:
thanks, this still feels like a bandaid but I don't really have an alternative solution on top of mind.  If the overall system is under stress we aren't going to help it by wasting cpu cycles so this should help a bit.

Reply:
I really appreciate your constructive advices, thanks.